.github/workflows/deploy-2-start.yml

```diff
@@ -114,20 +114,56 @@ jobs:
           spack env activate ${{ inputs.env-name }}
           spack --debug install --fresh ${{ vars.SPACK_INSTALL_PARALLEL_JOBS }} || exit $?
           spack module tcl refresh -y
+          EOT
+
+      - name: Get metadata from ${{ inputs.deployment-environment }}
+        env:
+          SPACK_ENV_PATH: ${{ steps.path.outputs.spack }}/var/spack/environments/${{ inputs.env-name }}
+        run: |
+          ssh ${{ secrets.USER}}@${{ secrets.HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
+          . ${{ steps.path.outputs.spack-config }}/spack-enable.bash
+          spack env activate ${{ inputs.env-name }}
 
-          # Obtain metadata
-          spack find --paths > ${{ steps.path.outputs.spack }}/var/spack/environments/${{ inputs.env-name }}/spack.location
-          spack find --format '{hash} {prefix}' | jq --raw-input --null-input '[inputs | split(" ") | {(.[0]): (.[1])}] | add' > ${{ steps.path.outputs.spack }}/var/spack/environments/${{ inputs.env-name }}/spack.location.json
+          spack find --paths > ${{ env.SPACK_ENV_PATH }}/spack.location
+          spack find --format '{hash} {prefix}' | jq --raw-input --null-input '[inputs | split(" ") | {(.[0]): (.[1])}] | add' > ${{ env.SPACK_ENV_PATH }}/spack.location.json
+
+          # Get the repos associated with the packages for the build database
+          jq -n '{}' > ${{ env.SPACK_ENV_PATH }}/build-db-pkgs.json
+          for pkg in ${{ vars.BUILD_DB_PACKAGES }}; do
+            # TODO: Is there a way to get the git attribute without concretizing?
+            pkg_repo_url=$(spack python -c "import spack.spec; print(spack.spec.Spec('$pkg').concretized().package.git)")
+
+            # We get the version of $pkg from spack.lock, and then strip
+            # potential 'git.' and '=VERSION' parts from 'git.TAG=VERSION'
+            version=$(jq --compact-output --raw-output \
+              --arg p "$pkg" \
+              '.concrete_specs | to_entries[].value | select(.name == $p) | .version
+                | match("^(?:git.)?([^=]*)")
+                | .captures[0].string' \
+              ${{ env.SPACK_ENV_PATH}}/spack.lock
+            )
```

Comment:
It doesn't have to happen in this PR, but this really looks like something that would benefit from being a function we could write some tests for. At the very least it would then be explicit about what is, and what is not, supported.

For reviewing this it would be helpful to show some output of this running on the command line. Even paste in an example `spack.lock` file so a reviewer could easily run some test code.

Reply:
The `spack.lock` I've been using for testing is the `access-esm1.5 2024.06.0` release. When I run the above I get the expected `access-esm1.5_2024.06.20`.